app/renderer/actions/Inspector.js

```diff
@@ -432,23 +432,46 @@ export function searchForElement (strategy, selector) {
 }
 
 
-export function resetSearchForPORootElement () {
+export function resetSearchForPOElements () {
   return (dispatch) => {
     dispatch({type: SEARCHING_FOR_ROOT_ELEMENTS});
   };
 }
 
-export function searchForPORootElement (strategy, selector, po) {
+export function searchForPOElements (po, strategyMap) {
+  const [type, selector] = Object.entries(po.selector)[0];
+  let strategy = strategyMap[type];
+
   return async (dispatch, getState) => {
     try {
       const callAction = callClientMethod({strategy, selector, fetchArray: true});
-      let {elements, variableName} = await callAction(dispatch, getState);
-      const findAction = findAndAssign(strategy, selector, variableName, true);
-      findAction(dispatch, getState);
-      elements = elements.map((el) => el.id);
+      let {elements} = await callAction(dispatch, getState);
       if (elements.length > 0) {
-        elements.push(po.root);
-        dispatch({type: SEARCHING_FOR_ROOT_ELEMENTS_COMPLETED, elements});
+        let passed = true;
+        if (po.elements) {
+          let po_elements = po.elements;
+          const cnt = po_elements.length;
+          if (cnt > 0) {
+            for (let i = 0; i < cnt; i++) {
```

Comment:
would prefer to use .forEach  instead of this c++ style iteration

Reply:
Here i need break the loop earlier once an element is not matched. forEach loop is not good for that purpose which is why the traditional for loop is used.

Reply:
Updated to use 'for of' loop.